File: C/selectionsort.c
#include <stdio.h>

int main()
{
    int n;
    int min;
    int temp;
    printf("Enter the number of elements in array\n");
    scanf("%d",&n);
    int arr[n];
    printf("Enter the elements of array\n");
    for (int i=0;i<n;i++)
    scanf("%d",&arr[i]);
    for(int i=0;i<=n-2;i++)
    {
        min=i;
        for (int j=i;j<=n-1;j++)
        {
            if (arr[j]<arr[min])
            min=j;
        }
        temp=arr[min];
        arr[min]=arr[i];
        arr[i]=temp;
    }
    for (int i=0;i<n;i++)
    printf("%d",arr[i]);
}

File: kotlin/selection_sort.kt
fun selectionSort(arr: IntArray) {
    val n = arr.size
    for (i in 0 until n - 1) {
        // Find the minimum element in unsorted array
        var minIndex = i
        for (j in i + 1 until n) {
            if (arr[j] < arr[minIndex]) {
                minIndex = j
            }
        }
        // Swap the found minimum element with the first element
        val temp = arr[i]
        arr[i] = arr[minIndex]
        arr[minIndex] = temp
    }
}

fun main() {
    print("Enter number of elements: ")
    val n = readln().toInt()

    val arr = IntArray(n)
    println("Enter elements:")
    for (i in 0 until n) {
        arr[i] = readln().toInt()
    }

    println("Original array: ${arr.joinToString(" ")}")
    selectionSort(arr)
    println("Sorted array:   ${arr.joinToString(" ")}")
}
